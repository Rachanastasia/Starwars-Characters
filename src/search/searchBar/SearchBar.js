import React from 'react';

export default function SearchBar(props) {

    const setTerm = e => props.term(e.target.value);

    return (
        <>
            <label htmlFor="search_bar"></label>
            <input type="text" name="search_bar" onChange={e => setTerm(e)} value={props.val} />
        </>
    )
}