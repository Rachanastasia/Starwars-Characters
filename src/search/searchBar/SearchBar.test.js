import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SearchBar from './SearchBar'




describe('SearchBar', () => {
    it('renders without crashign', () => {
        const div = document.createElement('div');
        ReactDOM.render(<SearchBar />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders UI as expected', ()=> {
        const tree = renderer
            .create(<SearchBar name='search' />)
            .toJSON();
            expect(tree).toMatchSnapshot();
    })







});