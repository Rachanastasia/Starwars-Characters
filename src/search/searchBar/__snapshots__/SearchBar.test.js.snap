// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders UI as expected 1`] = `
Array [
  <label
    htmlFor="search_bar"
  />,
  <input
    name="search_bar"
    type="text"
  />,
]
`;
